<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1148</width>
    <height>837</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color:rgb(209, 209, 209)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_6">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="topMargin">
       <number>20</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>50</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <property name="leftMargin">
         <number>10</number>
        </property>
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="maximumSize">
           <size>
            <width>520</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="cursor">
           <cursorShape>ArrowCursor</cursorShape>
          </property>
          <property name="mouseTracking">
           <bool>false</bool>
          </property>
          <property name="tabletTracking">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="styleSheet">
           <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: rgb(243, 243, 243); /* Use color names directly */
  border-radius: 20px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 10px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
          </property>
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="sizeConstraint">
              <enum>QLayout::SetDefaultConstraint</enum>
             </property>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <property name="spacing">
                <number>15</number>
               </property>
               <property name="topMargin">
                <number>70</number>
               </property>
               <property name="bottomMargin">
                <number>20</number>
               </property>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_4">
                 <property name="leftMargin">
                  <number>40</number>
                 </property>
                 <property name="rightMargin">
                  <number>40</number>
                 </property>
                 <item>
                  <widget class="QCheckBox" name="add_conjugates">
                   <property name="font">
                    <font>
                     <pointsize>12</pointsize>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="text">
                    <string>Add Conjugates</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <property name="topMargin">
                <number>20</number>
               </property>
               <property name="bottomMargin">
                <number>40</number>
               </property>
               <item>
                <widget class="QGroupBox" name="groupBox">
                 <property name="styleSheet">
                  <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: white; /* Use color names directly */
  border-radius: 15px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 5px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
                 </property>
                 <property name="title">
                  <string/>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_7">
                  <item row="0" column="0">
                   <widget class="PlotWidget" name="circle_graph">
                    <property name="styleSheet">
                     <string notr="true">QGraphicsView {
    background-color: white;
}</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="leftMargin">
                <number>20</number>
               </property>
               <property name="topMargin">
                <number>10</number>
               </property>
               <property name="rightMargin">
                <number>30</number>
               </property>
               <property name="bottomMargin">
                <number>20</number>
               </property>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_11">
                 <property name="leftMargin">
                  <number>50</number>
                 </property>
                 <item>
                  <widget class="QRadioButton" name="zero_radiobtn">
                   <property name="font">
                    <font>
                     <family>Calibri</family>
                     <pointsize>10</pointsize>
                     <bold>false</bold>
                    </font>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">    background-color: none</string>
                   </property>
                   <property name="text">
                    <string>ZERO</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_12">
                 <property name="leftMargin">
                  <number>50</number>
                 </property>
                 <item>
                  <widget class="QRadioButton" name="polae_radiobtn">
                   <property name="font">
                    <font>
                     <family>Calibri</family>
                     <pointsize>10</pointsize>
                    </font>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">    background-color:none
</string>
                   </property>
                   <property name="text">
                    <string>POLE</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <property name="spacing">
                <number>7</number>
               </property>
               <property name="topMargin">
                <number>20</number>
               </property>
               <property name="bottomMargin">
                <number>100</number>
               </property>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_8">
                 <property name="leftMargin">
                  <number>5</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="delete_zeros_btn">
                   <property name="minimumSize">
                    <size>
                     <width>100</width>
                     <height>40</height>
                    </size>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">QPushButton {
    width: 50%;
    background-color: #36393D; /* Replace with the corresponding color value for --dark-grey in your application */
    color: white;
    border-radius: 10px;
    text-align: center;

}

QPushButton:hover {
    background-color: #1b1d1f; /* Replace with the corresponding color value for --darkest-color in your application */
}
</string>
                   </property>
                   <property name="text">
                    <string>Delete All Zeros</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <widget class="QPushButton" name="delete_poles_btn">
                   <property name="minimumSize">
                    <size>
                     <width>100</width>
                     <height>40</height>
                    </size>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">QPushButton {
    width: 50%;
    background-color: #36393D; /* Replace with the corresponding color value for --dark-grey in your application */
    color: white;
    border-radius: 10px;
    text-align: center;

}

QPushButton:hover {
    background-color: #1b1d1f; /* Replace with the corresponding color value for --darkest-color in your application */
}
</string>
                   </property>
                   <property name="text">
                    <string>Delete All Poles</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_7">
                 <property name="rightMargin">
                  <number>5</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="delete_all_btn">
                   <property name="minimumSize">
                    <size>
                     <width>90</width>
                     <height>40</height>
                    </size>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">QPushButton {
    width: 50%;
    background-color: rgb(182, 13, 13); /* Replace with the corresponding color value for --dark-grey in your application */
    color: white;
    border-radius: 10px;
    text-align: center;

}

QPushButton:hover {
    background-color: white;
color:red /* Replace with the corresponding color value for --darkest-color in your application */
}
</string>
                   </property>
                   <property name="text">
                    <string>Delete All</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>950</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: rgb(243, 243, 243); /* Use color names directly */
  border-radius: 20px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 10px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
          </property>
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_14">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_7">
               <item>
                <widget class="QGroupBox" name="groupBox_2">
                 <property name="styleSheet">
                  <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: white; /* Use color names directly */
  border-radius: 15px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 5px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
                 </property>
                 <property name="title">
                  <string/>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_3">
                  <item row="0" column="0">
                   <widget class="PlotWidget" name="frequancy_responce_graph">
                    <property name="styleSheet">
                     <string notr="true">QGraphicsView {
    background-color: white;
}</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_8">
               <item>
                <widget class="QGroupBox" name="groupBox_6">
                 <property name="styleSheet">
                  <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: white; /* Use color names directly */
  border-radius: 15px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 5px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
                 </property>
                 <property name="title">
                  <string/>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_4">
                  <item row="0" column="0">
                   <widget class="PlotWidget" name="phase_responce_graph">
                    <property name="styleSheet">
                     <string notr="true">QGraphicsView {
    background-color: white;
}</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_13">
               <property name="leftMargin">
                <number>90</number>
               </property>
               <property name="rightMargin">
                <number>90</number>
               </property>
               <item>
                <widget class="QPushButton" name="all_pass_filter_btn">
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>40</height>
                  </size>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QPushButton {
    width: 50%;
    background-color: #36393D; /* Replace with the corresponding color value for --dark-grey in your application */
    color: white;
    border-radius: 10px;
    margin-left: auto;
    margin-right: auto;

}

QPushButton:hover {
    background-color: #1b1d1f; /* Replace with the corresponding color value for --darkest-color in your application */
}
</string>
                 </property>
                 <property name="text">
                  <string>ALL-Pass Filters</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_7">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>520</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: rgb(243, 243, 243); /* Use color names directly */
  border-radius: 20px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 10px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
        </property>
        <property name="title">
         <string/>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_18">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_11">
             <property name="spacing">
              <number>15</number>
             </property>
             <property name="topMargin">
              <number>70</number>
             </property>
             <property name="bottomMargin">
              <number>20</number>
             </property>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_16">
               <property name="leftMargin">
                <number>40</number>
               </property>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="Generate_sig_btn">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>40</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">QPushButton {
    width: 50%;
    background-color: #36393D; /* Replace with the corresponding color value for --dark-grey in your application */
    color: white;
    border-radius: 10px;
    text-align: center;

}

QPushButton:hover {
    background-color: #1b1d1f; /* Replace with the corresponding color value for --darkest-color in your application */
}
</string>
               </property>
               <property name="text">
                <string>Generate Signal</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_17">
               <property name="rightMargin">
                <number>40</number>
               </property>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_12">
             <property name="topMargin">
              <number>20</number>
             </property>
             <property name="bottomMargin">
              <number>10</number>
             </property>
             <item>
              <widget class="QWidget" name="hoverWidget" native="true">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>200</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>300</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: #36393D;  /* Use color names directly */
  border-radius: 15px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 5px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_8">
             <property name="styleSheet">
              <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: white; /* Use color names directly */
  border-radius: 15px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 5px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
             </property>
             <property name="title">
              <string/>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_15">
              <item>
               <widget class="PlotWidget" name="hoveredInput">
                <property name="styleSheet">
                 <string notr="true">QGraphicsView {
    background-color: white;
}</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="styleSheet">
              <string notr="true">QGroupBox {
  width: 65vw;
  height: 90vh;
  background-color: white; /* Use color names directly */
  border-radius: 15px;
  padding: 5px; /* Adjust padding as needed */
  margin-right: 5px;
}

/* You may need to adjust the sub-controls styles based on your specific requirements */
QGroupBox::title {
  subcontrol-origin: margin;
  subcontrol-position: top center;
  padding: 0 3px;
}

QGroupBox::indicator {
  width: 13px;
  height: 13px;
}


</string>
             </property>
             <property name="title">
              <string/>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="PlotWidget" name="hoveredOutput">
                <property name="styleSheet">
                 <string notr="true">QGraphicsView {
    background-color: white;
}</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1148</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
